src/lib/pages/frontsite/SelfCheckout/SelfCheckout.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import Overridable from 'react-overridable';
+import { Container, Header } from 'semantic-ui-react';
+import { BarcodeScanner } from '@components/BarcodeScanner';
+import { Html5Qrcode, Html5QrcodeSupportedFormats } from 'html5-qrcode';
+import { SelfCheckoutModal } from './SelfCheckoutModal';
+import { invenioConfig } from '@config';
+import _get from 'lodash/get';
+
+class SelfCheckout extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      scanner: null,
+      showModal: false,
+      barcode: null,
+    };
+  }
+
+  toggleModal = () => {
+    const { showModal, scanner } = this.state;
+    const shouldShowModal = !showModal && this.isItemLoanable();
+    this.setState({ showModal: shouldShowModal });
+    if (shouldShowModal) {
+      scanner.pause();
+    } else if (!scanner.isScanning) {
+      scanner.resume();
+    }
+  };
+
+  onScanSuccess = async (decodedText, decodedResult) => {
+    // Send to selfCheckout if different barcode
+    const { barcode } = this.state;
+    if (decodedText.trim() !== barcode) {
+      this.setState({ barcode: decodedText.trim() });
+
+      const { selfCheckOutSearch } = this.props;
+      await selfCheckOutSearch(decodedText.trim());
+      // open modal
+      this.toggleModal();
+    }
+  };
+
+  qrboxFunction = (viewfinderWidth, viewfinderHeight) => {
+    return {
+      width: viewfinderWidth * 0.9,
+      height: viewfinderWidth * 0.3,
+    };
+  };
```

Comment:
This function knows basically nothing where or how it is used, I would make it a bit more self-explanatory. You are also not using the Height, is that correct?
```suggestion
  getScannerArea = (cameraWidth, cameraHeight) => {
    return {
      width: cameraWidth * 0.9,  // 90% width
      height: cameraHeight * 0.3,  // 30% height
    };
  };
```

Reply:
Thanks, I will update the function! Also, yes I should use the height.